api/posts.js

```diff
@@ -0,0 +1,18 @@
+import { client } from "./axiosClient"
+
+export const getUserPosts = async (userId) => {
+  return client.get(`/posts?userId=${userId}`).then(response => response.data);
+}
+
+export const createPost = async (data) => {
+  return client.post('/posts', data);
+}
+
+export const updatePost = async (data) => {
+  console.log(data);
+  return client.patch(`/posts/${data.id}`, data);
```

Comment:
The `updatePost` function logs the `data` to the console, which might be unnecessary for production code. Consider removing the `console.log(data);` line if it's not needed for debugging purposes.